### PR TITLE
ref(debug-files-settings): Add settings dialog for appstore connect

### PR DESCRIPTION
**Pull request by priscilawebdev**
Add a new custom repository option in debug files. The new option is called "App Store Connect"

image

**Previews:**

Adding Mode

image

Warnings when the validation fails

image

Editing Mode

image

**Why do we need apple store connection credentials + apple authentication?**

Currently, developers have to make sure they downloaded the `dSYMs` from App Store Connect in XCode before calling `sentry-cli `and[ uploading their debug symbols](https://docs.sentry.io/platforms/apple/dsym/) to Sentry.

There is also a Fastlane action which can download automate the downloading symbols and uploading to sentry. This works well once setup on CI but requires the entire pipeline to keep running until all the outputs are built by Apple.  Furthermore, Apple can later re-build the bitcode with new optimisations or for new target platforms and the resulting symbols would need to be downloaded again to the developer's system before uploading back to Sentry.

App Store Connect has an OpenAPI which can be used. Unfortunately as of version 1.2 this does not yet include access to the `dSYMs` of builds, so we will need to scrape the[ UI like Fastlane does](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/download_dsyms.rb).

> Fastlane is already in contact with Apple and have plans to include this official API, see [https://github.com/fastlane/fastlane/discussions/17485#discussioncomment-125619](https://github.com/fastlane/fastlane/discussions/17485#discussioncomment-125619)

Please check this page for more details https://www.notion.so/sentry/App-Store-Connect-Overview-ea53e47ee5724e2a8c724df39a2698f7

ps: everything is under the feature flag `organizations:app-store-connect`

### CODE REVIEW

File: static/app/utils/theme.tsx [hunk 1]
```diff
@@ -600,7 +614,7 @@ const darkAliases = {
   overlayBackgroundAlpha: 'rgba(18, 9, 23, 0.7)',
   tagBarHover: colors.purple300,
   tagBar: colors.gray400,
-  businessIconColors: [colors.pink100, colors.pink300],
+  badgeText: colors.black,
```
Comment (matejminar) [inline static/app/utils/theme.tsx:617]: Did we remove `businessIconColors` on purpose?

File: static/app/views/settings/projectDebugFiles/index.tsx [hunk 2]
```diff
@@ -147,6 +178,55 @@ class ProjectDebugSymbols extends AsyncView<Props, State> {
     });
   }
 
+  getFieldsSymbolSources() {
+    const {organization} = this.props;
+
+    if (
+      !organization.features?.includes('app-store-connect') ||
+      this.getAppStoreConnectId()
+    ) {
```
Comment (matejminar) [inline static/app/views/settings/projectDebugFiles/index.tsx:187]: Do we need to guard this whole condition by feature flag?
Also the `renderAppStoreConnectAlert`.

Could it be that someone already has the app store id and then we decide for some reason to turn off the feature flag?

**Review by matejminar — APPROVED** [review]
LGTM

File: static/app/utils/theme.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/utils/theme.tsx:617]: oh no... good catch... let me put it back

**Pull request merged**